compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -175,13 +176,20 @@ fn program() -> impl NoirParser<ParsedModule> {
 ///       | %empty
 fn module() -> impl NoirParser<ParsedModule> {
     recursive(|module_parser| {
-        empty()
-            .to(ParsedModule::default())
-            .then(spanned(top_level_statement(module_parser)).repeated())
-            .foldl(|mut program, (statement, span)| {
-                if let Some(kind) = statement.into_item_kind() {
-                    program.items.push(Item { kind, span });
-                }
+        inner_attributes()
+            .then(
+                empty()
+                    .to(ParsedModule::default())
```

Comment:
I think we can be more flexible than rust here and allow parsing these as a normal top level statement. Notably this could allow other attributes to output an inner attribute.

Reply:
Sounds good, I also hesitated about this approach or the other. I saw Rust only allows them at the top, though in reality it also allows them syntactically anywhere but it gives an error... I'll change it to a TopLevelItem, there's also something I forgot to mention which is that these inner attributes sometimes disappear when the file gets formatted (only if the module is empty) so moving them to a TopLevelItem might fix that too 🤞